src/Build/Collections/RetrievableEntryHashSet/HashSet.cs

```diff
@@ -1,219 +1,142 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
difficult to review (will be easier when other PR is merged)

essentially the dotnet/runtime version, with the MSBuild modifications, unused stuff removed, some style cleanup.

note -- we're probably a bit inconsistent about verifythrow vs debug.assert. it might be good at some future point to figure out what is actually exposed publicly. everything else should be debug.assert. and we can probably stub out some of the interface implementations if they're not publicly exposed.

Reply:
and maybe version can become debug only too.

Reply:
removed version.